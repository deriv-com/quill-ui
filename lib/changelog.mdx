import { Meta, Markdown } from "@storybook/blocks";
import Changelog from "../CHANGELOG.md?raw";

<Meta title="Changelog" />

# Changelog

<Markdown>{Changelog}</Markdown>
