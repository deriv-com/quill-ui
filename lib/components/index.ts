export * from "./ActionSheet";
export * from "./Atom";
export * from "./Button";
export * from "./Badge";
export * from "./Chip";
export * from "./Input";
export * from "./Link";
export * from "./Modal";
export * from "./Notification";
export * from "./Pagination";
export * from "./SectionMessage";
export * from "./SegmentedControl";
export * from "./SelectionControl";
export * from "./Snackbar";
export * from "./Tab";
export * from "./Tag";
export * from "./Tooltip";
export * from "./Typography";
export * from "./Loader";
export * from "./QuillResetCssWrapper";
