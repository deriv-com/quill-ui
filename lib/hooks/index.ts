export * from "./useNotifications";
export * from "./useTheme";
export * from "./useSwipeBlock";
