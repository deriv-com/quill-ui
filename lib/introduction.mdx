import { Meta, Markdown } from "@storybook/addon-docs";
import Readme from "../readme.md?raw";

<Meta title="Introduction" />

<Markdown>{Readme}</Markdown>
