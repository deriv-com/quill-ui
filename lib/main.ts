export * from "./components";
export * from "./hooks/useNotifications";
export * from "./hooks/useTheme";
export * from "./hooks/usePaginationRange";
export * from "./hooks/useSnackbar";
export * from "./providers/theme/themeProvider";
export * from "./providers/snackbar/snackbarProvider";
export * from "./providers/notification/notificationsProvider";
