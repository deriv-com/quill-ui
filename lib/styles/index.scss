/* Global Styles */
@import "./quill.scss";

@at-root {
    @import url("https://fonts.googleapis.com/css2?family=Ubuntu:wght@700&display=swap");
    @import url("https://fonts.googleapis.com/css2?family=IBM+Plex+Sans:ital,wght@0,400;0,700;1,400;1,700&display=swap");
    @import url("https://fonts.googleapis.com/css2?family=IBM+Plex+Mono:ital,wght@0,400;0,700;1,400;1,700&display=swap");
}

.quill {
    &__background {
        @each $bg in primary, secondary {
            &--#{$bg} {
                @each $type in base, container {
                    &__type--#{$type} {
                        background-color: var(
                            --semantic-color-background-#{$bg}-#{$type}
                        );
                    }
                }
            }
        }

        @each $bg in dialog, overlay {
            &--#{$bg} {
                background-color: var(--semantic-color-background-#{$bg});
            }
        }

        &--overflow__number {
            @each $number in 1, 2 {
                &--#{$number} {
                    @each $position in top, right, bottom, left {
                        background-color: var(
                            --semantic-color-background-overflow-#{$number}-#{$position}
                        );
                    }
                }
            }
        }
    }
}

.quill {
    &__color {
        &--primary-coral {
            background-color: var(--core-color-solid-coral-700);
            color: var(--core-color-solid-slate-50);
            fill:  var(--core-color-solid-slate-50);
            border: none;
            &:hover:not(:disabled){
                background-color: var(--core-color-solid-coral-800);
            }
            &:active:not(:disabled){
                background-color: var(--core-color-solid-coral-900);
            }
            &:disabled{
                opacity: 0.4;
            }
        }
        &--primary-black{
            background-color: var(--core-color-solid-slate-1400);
            color: var(--core-color-solid-slate-50);
            fill: var(--core-color-solid-slate-50);
            border: none;
            &:hover:not(:disabled){
                background-color: var(--core-color-solid-slate-700);
            }
            &:active:not(:disabled){
                background-color: var(--core-color-opacity-black-800);
            }
            &:disabled{
                opacity: 0.48;
              
            }
        }
        &--primary-white{
            background-color: var(--core-color-solid-slate-50);
            color: var(--core-color-solid-slate-1400);
            fill: var(--core-color-solid-slate-1400);
            border: none;
            &:hover:not(:disabled){
                background-color: var(--core-color-opacity-white-800);
            }
            &:active:not(:disabled){
                background-color: var(--core-color-opacity-white-700);
            }
            &:disabled{
                background-color: var(--core-color-solid-slate-50);
            }
        }
        &--primary-purchase {
            background-color : var(--core-color-solid-emerald-700);
            color: var(--core-color-solid-slate-50);
            fill: var(--core-color-solid-slate-50);
            border: none;
            &:hover:not(:disabled) {
                background-color: var(--core-color-solid-emerald-800);
            }
            &:active:not(:disabled) {
                background-color: var(--core-color-solid-emerald-900);
            }
            &:disabled{
                opacity: 0.48;
            }
        }
        &--primary-sell {
            background-color: var(--core-color-solid-cherry-700);
            color: var(--core-color-solid-slate-50);
            fill: var(--core-color-solid-slate-50);
            border: none;
            &:hover:not(:disabled) {
                background-color: var(--core-color-solid-cherry-800);
            }
            &:active:not(:disabled) {
                background-color: var(--core-color-solid-cherry-900);
            }
            &:disabled{
                opacity: 0.48;
            }
        }
        &--secondary-coral {
            background-color :transparent;
            border-style: solid;
            border-color: var(--core-color-solid-coral-700);
            color:  var(--core-color-solid-coral-700);
            fill: var(--core-color-solid-coral-700);
            &:hover:not(:disabled){
                background-color: var(--core-color-opacity-coral-100);
            }
            &:active:not(:disabled){
                background-color: var(--core-color-opacity-coral-200);
            }
            &:disabled{
                opacity: 0.4;
            }
        }
        &--secondary-black{
            background-color:transparent;
            border-style: solid;
            border-color: var(--core-color-solid-black-1400);
            color:  var(--core-color-solid-black-1400);
            fill:  var(--core-color-solid-black-1400);
            &:hover:not(:disabled){
                background-color: var(--core-color-opacity-black-100);
            }
            &:active:not(:disabled){
                background-color: var(--core-color-opacity-black-200);
            }
            &:disabled{
                opacity: 0.4;
            }
        }
        &--secondary-white{
            background-color:transparent;
            border-style: solid;
            border-color: var(--core-color-solid-slate-50);
            color:  var(--core-color-solid-slate-50);
            fill:  var(--core-color-solid-slate-50);
            &:hover:not(:disabled){
                background-color: var(--core-color-opacity-white-200);
            }
            &:active:not(:disabled){
                background-color: var(--core-color-opacity-white-100);
            }
            &:disabled{
                opacity: 0.4;
            }
        }
        &--secondary-purchase {
            background-color :transparent;
            border-style: solid;
            border-color: var(--core-color-solid-emerald-700);
            color: var(--core-color-solid-emerald-700);
            fill: var(--core-color-solid-emerald-700);
            &:hover:not(:disabled) {
                background-color:var(--core-color-solid-emerald-100);
            }
            &:active:not(:disabled) {
                background-color: var(--core-color-opacity-emerald-200);
            }
            &:disabled{
                opacity: 0.48;
            }
        }
        &--secondary-sell{
            background-color :transparent;
            border-style: solid;
            border-color: var(--core-color-solid-cherry-700);
            color: var(--core-color-solid-cherry-700);
            fill: var(--core-color-solid-cherry-700);
            &:hover:not(:disabled) {
                background-color: var(--core-color-solid-cherry-100);
            }
            &:active:not(:disabled) {
                background-color: var(--core-color-opacity-cherry-200);
            }
            &:disabled{
                opacity: 0.48;
            }
        }
        &--tertiary-coral{
            background-color :transparent;
            color: var(--core-color-solid-coral-700);
            fill: var(--core-color-solid-coral-700);
            text-decoration: underline;
            border: none;
            &:hover:not(:disabled){
                background-color: var(--core-color-opacity-coral-100);
            }
            &:active:not(:disabled){
                background-color: var(--core-color-opacity-coral-200);
            }
            &:disabled{
                opacity: 0.4;
            }
        }
        &--tertiary-black{
            background-color :transparent;
            color: var(--core-color-solid-black-1400);
            fill: var(--core-color-solid-black-1400);
            text-decoration: underline;
            border: none;
            &:hover:not(:disabled){
                background-color: var(--core-color-opacity-black-100);
            }
            &:active:not(:disabled){
                background-color: var(--core-color-opacity-black-200);
            }
            &:disabled{
                opacity: 0.48;
            }
        }
        &--tertiary-white{
            background-color :transparent;
            color: var(--core-color-solid-slate-50);
            fill: var(--core-color-solid-slate-50);
            text-decoration: underline;
            border: none;
            &:hover:not(:disabled){
                background-color: var(--core-color-opacity-white-200);
            }
            &:active:not(:disabled){
                background-color: var(--core-color-opacity-white-100);
            }
            &:disabled{
                opacity: 0.48;
            }
        }
        &--tertiary-purchase{
            background-color :transparent;
            color: var(--core-color-solid-emerald-700);
            fill: var(--core-color-solid-emerald-700);
            text-decoration: underline;
            border: none;
            &:hover:not(:disabled) {
                background-color: var(--core-color-opacity-emerald-100);
            }
            &:active:not(:disabled) {
                background-color: var(--core-color-opacity-emerald-200);
            }
            &:disabled{
                opacity: 0.48;
            }
        }
        &--tertiary-sell{
            background-color :transparent;
            color: var(--core-color-solid-cherry-700);
            fill: var(--core-color-solid-cherry-700);
            text-decoration: underline;
            border: none;
            &:hover:not(:disabled) {
                background-color: var(--core-color-solid-cherry-100);
            }
            &:active:not(:disabled) {
                background-color: var(--core-color-opacity-cherry-200);
            }
            &:disabled{
                opacity: 0.48;
            }
        }
   
    }

}
