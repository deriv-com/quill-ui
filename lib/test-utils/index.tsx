import React, { ReactElement } from "react";
import {
    queries,
    Queries,
    render,
    RenderHookOptions,
    RenderHookResult,
    RenderOptions,
    renderHook,
} from "@testing-library/react";
import BreakpointProvider from "@providers/breakpoint/breakpointProvider";
import ThemeProvider from "@providers/theme/themeProvider";

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
    return (
        <BreakpointProvider>
            <ThemeProvider theme="light">{children}</ThemeProvider>
        </BreakpointProvider>
    );
};

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: AllTheProviders, ...options });

function customRenderHook<
    Result,
    Props,
    Q extends Queries = typeof queries,
    Container extends Element | DocumentFragment = HTMLElement,
    BaseElement extends Element | DocumentFragment = Container,
>(
    render: (initialProps: Props) => Result,
    options?: Omit<
        RenderHookOptions<Props, Q, Container, BaseElement>,
        "wrapper"
    >,
): RenderHookResult<Result, Props> {
    return {
        ...renderHook(render, { wrapper: AllTheProviders, ...options }),
    };
}

export * from "@testing-library/react";
export { customRender as render };
export { customRenderHook as renderHook };
