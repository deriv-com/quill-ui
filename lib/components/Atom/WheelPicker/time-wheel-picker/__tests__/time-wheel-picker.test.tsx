import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import TimeWheelPickerContainer from "..";
import { TimeWheelPickerContainerProps } from "lib/main";

let mockSetSelectedTime = jest.fn();
let mockClose = jest.fn();
let mockSetSelectedValue = jest.fn();

const renderComponent = (args: TimeWheelPickerContainerProps) =>
    render(
        <div
            style={{
                height: "240px",
            }}
        >
            <TimeWheelPickerContainer
                {...args}
                close={mockClose}
                setSelectedValue={mockSetSelectedValue}
                setSelectedTime={mockSetSelectedTime}
            />
            ,
        </div>,
    );
describe("GenericTimeWheelPickerContainer", () => {
    global.ResizeObserver = class {
        observe() {}
        unobserve() {}
        disconnect() {}
    };

    HTMLElement.prototype.scrollIntoView = jest.fn();

    beforeEach(() => {
        mockClose = jest.fn();
        mockSetSelectedTime = jest.fn();
        mockSetSelectedValue = jest.fn();
    });

    it("renders correctly", () => {
        const { asFragment } = renderComponent({ is12Hour: true });
        expect(asFragment()).toMatchSnapshot();
    });

    it("handles ArrowRight and ArrowLeft keydown events correctly", () => {
        const { getAllByRole } = renderComponent({ is12Hour: true });
        const wheelPickers = getAllByRole("listbox");

        fireEvent.keyDown(wheelPickers[0], { key: "ArrowRight" });
        expect(mockSetSelectedTime).toHaveBeenCalledTimes(3);

        fireEvent.keyDown(wheelPickers[1], { key: "ArrowLeft" });
        expect(mockSetSelectedTime).toHaveBeenCalledTimes(3);

        fireEvent.keyDown(wheelPickers[2], { key: "Escape" });
        expect(mockClose).toHaveBeenCalledTimes(0);
    });

    it("should select elements on scroll", async () => {
        const { getAllByRole } = renderComponent({ is12Hour: true });
        const wheelPickers = getAllByRole("listbox");

        waitFor(() => {
            fireEvent.scroll(wheelPickers[0], { target: { scrollTop: 48 } });
        });
        expect(mockSetSelectedTime).toHaveBeenNthCalledWith(4, "2:0 AM");
    });

    it("should render in 24 hour format", async () => {
        const { getAllByRole } = renderComponent({ is12Hour: false });
        const wheelPickers = getAllByRole("listbox");
        expect(wheelPickers.length).toBe(2);
    });
});
