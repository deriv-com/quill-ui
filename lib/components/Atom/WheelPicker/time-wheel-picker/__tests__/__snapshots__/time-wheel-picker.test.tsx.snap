// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericTimeWheelPickerContainer renders correctly 1`] = `
<DocumentFragment>
  <div
    style="height: 240px;"
  >
    <div
      class="quill-wheel-picker"
    >
      <div
        class="quill-wheel-picker__container"
        style="height: 100%;"
      >
        <ul
          class="quill-wheel-picker__data-items quill-wheel-picker__data-items-left"
          role="listbox"
        >
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              01
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              02
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              03
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              04
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              05
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              06
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              07
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              08
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              09
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              10
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              11
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--false quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label"
            >
              12
            </p>
          </li>
        </ul>
      </div>
      <div
        class="quill-wheel-picker__container"
        style="height: 100%;"
      >
        <ul
          class="quill-wheel-picker__data-items quill-wheel-picker__data-items-center quill-wheel-picker__data-items__minute"
          role="listbox"
        >
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--false quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label"
            >
              00
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              02
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              04
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              06
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              08
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              10
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              12
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              14
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              16
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              18
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              20
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              22
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              24
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              26
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              28
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              30
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              32
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              34
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              36
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              38
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              40
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              42
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              44
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              46
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              48
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              50
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              52
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              54
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              56
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              58
            </p>
          </li>
        </ul>
      </div>
      <div
        class="quill-wheel-picker__container"
        style="height: 100%;"
      >
        <ul
          class="quill-wheel-picker__data-items quill-wheel-picker__data-items-right"
          role="listbox"
        >
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--false quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label"
            >
              AM
            </p>
          </li>
          <li
            class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-center quill-dropdown-item__selected--false__disabled--true quill-wheel-picker__data-item"
            role="option"
            tabindex="0"
          >
            <p
              class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label quill-typography__color--disabled"
            >
              PM
            </p>
          </li>
        </ul>
      </div>
    </div>
    ,
  </div>
</DocumentFragment>
`;
